src/shader/AstGen.zig

```diff
@@ -2945,10 +2967,10 @@ fn genGenericFloatTripleBuiltin(
                 .op = op,
                 .result_type = a1_res,
                 .a1_type = a1_res,
-                .a2_type = a1_res,
+                .a2_type = a2_res,
```

Comment:
@alichraghi - should this be a1 because of the coerce? The instruction below was wrong but not sure about this one.

Reply:
that looks correct. i have made a typo